packages/cosmic-swingset/test/test-make.js

```diff
@@ -1,45 +1,249 @@
+        runMake([...bind({ ACCT_ADDR }), 'provision-acct']),
+      ),
+  });
+};
+
+test.before(async t => {
+  const filename = new URL(import.meta.url).pathname;
+  const dirname = ambientPath.dirname(filename);
+  const makefileDir = ambientPath.join(dirname, '..');
+
+  const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+  const pspawn =
+    bin =>
+    (args = [], opts = {}) => {
+      let child;
+      const exit = new Promise((resolve, reject) => {
+        // console.debug('spawn', bin, args, { cwd: makefileDir, ...opts });
+        child = ambientSpawn(bin, args, { cwd: makefileDir, ...opts });
+        child.addListener('exit', code => {
+          if (code !== 0) {
+            reject(Error(`exit ${code} from: ${bin} ${args}`));
+            return;
+          }
+          resolve(0);
+        });
+      });
+      return { child, exit };
+    };
+
+  const pspawnMake = pspawn('make');
+  const pspawnAgd = pspawn('bin/ag-chain-cosmos');
+  const scenario2 = makeScenario2({ pspawnMake, pspawnAgd, delay, log: t.log });
+  await scenario2.setup();
+  const soloAddr = await ambientFs.promises
+    .readFile('./t1/8000/ag-cosmos-helper-address', 'utf-8')
+    .then(line => line.trim());
+  const bootstrapAddr = await ambientFs.promises
+    .readFile('./t1/bootstrap-address', 'utf-8')
+    .then(line => line.trim());
+  // console.debug('scenario2 addresses', { soloAddr, bootstrapAddr });
+
+  t.context = { scenario2, pspawnAgd, bootstrapAddr, soloAddr };
+});
+
+test.serial('make and exec', async t => {
+  const { pspawnAgd, scenario2 } = t.context;
+  t.log('exec agd');
+  t.is(await pspawnAgd([]).exit, 0, 'exec agd exits successfully');
+  t.log('run chain to halt');
+  t.is(
+    await scenario2.runToHalt(),
+    0,
+    'make scenario2-run-chain-to-halt is successful',
   );
-  await new Promise(resolve =>
-    spawn('make', ['scenario2-run-chain-to-halt'], {
-      cwd: `${dirname}/..`,
-      stdio: ['ignore', 'ignore', 'inherit'],
-    }).addListener('exit', code => {
-      t.is(code, 0, 'make scenario2-run-chain-to-halt is successful');
-      resolve();
-    }),
+  t.log('resume chain and halt');
+  t.is(
+    await scenario2.runToHalt(),
+    0,
+    'make scenario2-run-chain-to-halt succeeds again',
   );
-  await new Promise(resolve =>
-    spawn('bin/ag-chain-cosmos', ['export', '--home=t1/n0'], {
-      cwd: `${dirname}/..`,
-      stdio: ['ignore', 'ignore', 'inherit'],
-    }).addListener('exit', code => {
-      t.is(code, 0, 'export exits successfully');
-      resolve();
+  t.log('export');
+  t.is(await scenario2.export(), 0, 'export exits successfully');
+});
+
+// SKIP: struggling with timing issues resulging in one of...
+// Error: cannot grab 250000uist coins: 0uist is smaller than 250000uist: insufficient funds
+// error: code = NotFound desc = account agoric1mhu... not found
+// Sometimes I can get this test to work alone, but not
+// if run with the test above.
```

Comment:
Seems tackling someday. If you agree, this would be a good place to reference an issue for it

Reply:
ok: #6766